pandas/core/arrays/datetimes.py

```diff
@@ -1368,8 +1368,8 @@ def date(self):
         'weekday_name',
         "The name of day in a week (ex: Friday)\n\n.. deprecated:: 0.23.0")
 
-    dayofyear = _field_accessor('dayofyear', 'doy',
-                                "The ordinal day of the year.")
+    day_of_year = _field_accessor('day_of_year', 'doy',
```

Comment:
we need to preserve the original with a deprecation warning

Reply:
I will add deprecation warnings on the weekend. Thank you for your feedback.